### PR TITLE
Paul90/restrict

### PR DESCRIPTION
**Pull request by WardCunningham**
First step towards login-to-view.
http://ward.asia.wiki.org/login-to-view.html

- We accept --restricted parameter with no arguments to mean restricted to owner only.
- We replace all .json requests with empty page titled "Login Required".

Login, logout and claim work to some degree. This is a foundation for improvements:

- [x] more specific restrictions
- [x] more careful security analysis
- [x] improve user experience

### CODE REVIEW

File: server/social.coffee [hunk 1]
```diff
@@ -387,6 +387,75 @@ module.exports = exports = (log, loga, argv) ->
       }
       res.render(path.join(__dirname, '..', 'views', 'done.html'), info)
 
+    # if configured, enforce restricted access to json
+    if argv.restricted?
+
+      loginToView = (req) ->
+        # things = [ { value: 'ward.cunningham@gmail.com', type: 'account' } ]
+        allowed = [
```
Comment (paul90) [inline server/social.coffee]: The list of allowed email domains could be loaded from a config item, not sure of the best name but something like `allowed_domains`. So config for the hardcoded example would be something like:

``` json
  "wikiDomains": {
    "example.wiki": {
      "restricted": true,
      "allowed_domains": [ "dayton.k12.or.us", "stu.dayton.k12.or.us",
          "newberg.k12.or.us", "stu.newberg.k12.or.us",
          "innovateoregon.org",
          "newrelic.com",
          "c2.com",
          "andysylvester.com"
      ]
    }
  }
```

So, line 395 would become:
``` coffeescript
        if argv.allowed_domains? and Array.isArray(argv.allowed_domains)
          allowed = argv.allowed_domains
        else
          allowed = []
```

Comment (WardCunningham) [reply server/social.coffee]: Well, this looks really simple. But how does allowed_domains get into argv?

**Comment by WardCunningham** [comment]
This pull request is ready to be merged.

**Comment by paul90** [comment]
Published as + wiki-security-passportjs@0.2.0

**Pull request merged**